api/validate.js

```diff
@@ -0,0 +1,49 @@
+const Validator = require('validatorjs');
+const validator = async (body, rules, customMessages, callback) => {
+    const validation = new Validator(body, rules, customMessages);
+    validation.passes(() => callback(null, true));
+    validation.fails(() => callback(validation.errors, false));
+};
+
+const titleEnglish =  /^[a-zA-Z\s]*$/;
+Validator.register('EnglishLetters', value => titleEnglish.test(value),
+    'the title mast to be with ב or spaces Dipulatives');
+
+    const description =  /^[a-zA-Z\s]*$/;
+
+// Tighten password policy
+Validator.register('desEnglishLetters', value => description.test(value),
+    'the description mast to be with English letters or spaces Dipulatives');
+
+    const tapedomin = /^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\.)+[A-Za-z]{2,6}$/;
+
+    // Tighten password policy
+    Validator.register('tapedomin', value => tapedomin.test(value),
+        'the tapedomin not Right');
+    
+
+    const dns = require('dns');
+
+    function isDomainAvailable(domain) {
+      return new Promise((resolve, reject) => {
+        dns.resolve(domain, (err) => {
+          if (err && err.code === 'ENOTFOUND') {
+            resolve(true); // domain is available
+          } else {
+            resolve(false); // domain is not available
+          }
+        });
+      });
+    }
+
+    Validator.register('isDomainAvailable', async (value, args) => {
+        const domain = value.trim();
+        const available = await isDomainAvailable(domain);
+        return available;
+      }, 'The domain is not available.');
+    const allowedCpuTypes = [686,586,486,386  ];
```

Comment:
save this array in a variable, and use it every where it needed.

Reply:
זה בידיוק מה שעשיתי

Reply:
יש פה רווחים מיותרים
וכדאי שתשתמשי בarray בשביל הכיתוב בסטרינג